platform/macos/app/MapDocument.m

```diff
@@ -185,8 +184,11 @@ - (IBAction)takeSnapshot:(id)sender {
     options.zoomLevel = self.mapView.zoomLevel;
     
     // Create and start the snapshotter
-    snapshotter = [[MGLMapSnapshotter alloc] initWithOptions:options];
-    [snapshotter startWithCompletionHandler:^(MGLMapSnapshot *snapshot, NSError *error) {
+    __weak __typeof__(self) weakSelf = self;
+    _snapshotter = [[MGLMapSnapshotter alloc] initWithOptions:options];
+    [_snapshotter startWithCompletionHandler:^(MGLMapSnapshot *snapshot, NSError *error) {
+        __typeof__(self) strongSelf = weakSelf;
```

Comment:
Return early before dereferencing `strongSelf` below. It’s definitely possible to close the window before the snapshot is complete, thereby causing a crash.